processes/processes_test.go

```diff
@@ -179,63 +183,176 @@ func TestCollectMetrics(t *testing.T) {
+				So(len(results), ShouldEqual, 2)
+
+				for _, r := range results {
+					ns := r.Namespace
+					So(mockProc.validateValue(ns[len(ns)-1].Value, r.Data.(uint64)), ShouldBeTrue)
+				}
+			})
+
+			Convey("check process instance count with process name specified", func() {
+				results, err := procPlugin.CollectMetrics([]plugin.Metric{
+					plugin.Metric{
+						Namespace: plugin.NewNamespace("intel", "procfs", "processes", "process", "fake", "ps_count"),
+						Config:    cfg,
+					},
+					plugin.Metric{
+						Namespace: plugin.NewNamespace("intel", "procfs", "processes", "process", "NetworkManager", "ps_count"),
+						Config:    cfg,
+					},
+				})
+
+				So(err, ShouldBeNil)
+				So(len(results), ShouldEqual, 2)
+
+				matchedNs := 0
+				for _, r := range results {
+					ns := strings.Join(r.Namespace.Strings(), "/")
+
+					switch ns {
+					case "intel/procfs/processes/process/fake/ps_count":
+						matchedNs++
+						So(r.Data, ShouldEqual, 2)
+					case "intel/procfs/processes/process/NetworkManager/ps_count":
+						matchedNs++
+						So(r.Data, ShouldEqual, 1)
+					}
+				}
+
+				So(matchedNs, ShouldEqual, 2)
+			})
+
+			Convey("check process instance count with process name equal to asterisk", func() {
+				results, err := procPlugin.CollectMetrics([]plugin.Metric{
+					plugin.Metric{
+						Namespace: plugin.NewNamespace("intel", "procfs", "processes", "process").
 							AddDynamicElement("process_name", "process name").
 							AddStaticElement("ps_count"),
-						Config_: cfg.ConfigDataNode,
+						Config: cfg,
 					},
 				})
 
 				So(err, ShouldBeNil)
-				// 2 metrics exposed by processes
 				So(len(results), ShouldEqual, 2)
 
+				matchedNs := 0
 				for _, r := range results {
-					ns := r.Namespace()
-					So(mockProc.validateValue(ns[len(ns)-1].Value, r.Data().(uint64)), ShouldBeTrue)
+					ns := strings.Join(r.Namespace.Strings(), "/")
+
+					switch ns {
+					case "intel/procfs/processes/process/fake/ps_count":
+						matchedNs++
+						So(r.Data, ShouldEqual, 2)
+					case "intel/procfs/processes/process/NetworkManager/ps_count":
+						matchedNs++
+						So(r.Data, ShouldEqual, 1)
+					}
 				}
+
+				So(matchedNs, ShouldEqual, 2)
 			})
 
 			Convey("when names of collect metrics include asterisk", func() {
-				mockMtsWithAsterisk := append(mockMts, plugin.MetricType{
-					Namespace_: core.NewNamespace("intel", "procfs", "processes").
+				mockMtsWithAsterisk := append(mockMts, plugin.Metric{
+					Namespace: plugin.NewNamespace("intel", "procfs", "processes", "process").
 						AddDynamicElement("process_name", "process name").
-						AddStaticElement("ps_count"),
+						AddDynamicElement("process_pid", "process identificator").
+						AddStaticElement("ps_data"),
 				})
 
 				results, err := procPlugin.CollectMetrics(mockMtsWithAsterisk)
 
 				So(err, ShouldBeNil)
 				// 2 dynamic metrics exposed by processes + 10 status metrics defined in mockMts
```

Comment:
Comment is not appropriate to the current state, now there are 15 metrics instead of 12.